src/main/java/com/books/services/BookService.java

```diff
@@ -33,7 +41,12 @@
     }
 
     public int generateId() {
```

Comment:
This is a very strange method. Do you know about auto generated id in RDBMS? https://www.w3schools.com/sql/sql_autoincrement.asp